x-pack/lib/filters/geoip/database_metric.rb

```diff
@@ -0,0 +1,74 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+require "logstash/util/loggable"
+require_relative "util"
+require_relative "database_metadata"
+require_relative "download_manager"
+require "date"
+require "time"
+
+module LogStash module Filters module Geoip class DatabaseMetric
+  include LogStash::Util::Loggable
+  include LogStash::Filters::Geoip::Util
+
+  DATABASE_INIT = "init"
+  DATABASE_UP_TO_DATE = "up_to_date"
+  DATABASE_TO_BE_EXPIRED = "to_be_expired"
+  DATABASE_EXPIRED = "expired"
+
+  DOWNLOAD_SUCCEEDED = "succeeded"
+  DOWNLOAD_FAILED = "failed"
+  DOWNLOAD_UPDATING = "updating"
```

Comment:
It looks like the code all references these by constant now, which is a great improvement.

Since ruby strings are mutable, symbols would maintain readability while being marginally more safe:

```suggestion
  DATABASE_INIT = :init
  DATABASE_UP_TO_DATE = :up_to_date
  DATABASE_TO_BE_EXPIRED = :to_be_expired
  DATABASE_EXPIRED = :expired

  DOWNLOAD_SUCCEEDED = :succeeded
  DOWNLOAD_FAILED = :failed
  DOWNLOAD_UPDATING = :updating
```

Reply:
Upon second thought, after re-reading the initial PR, using a symbol would probably put us back in the `A gauge metric of an unknown type (org.jruby.RubySymbol) has been created for key: status` state, yeah? Perhaps we could use frozen strings instead?

```suggestion
  DATABASE_INIT = "init".freeze
  DATABASE_UP_TO_DATE = "up_to_date".freeze
  DATABASE_TO_BE_EXPIRED = "to_be_expired".freeze
  DATABASE_EXPIRED = "expired".freeze

  DOWNLOAD_SUCCEEDED = "succeeded".freeze
  DOWNLOAD_FAILED = "failed".freeze
  DOWNLOAD_UPDATING = "updating".freeze
```

Reply:
exactly, I should freeze it